tests/bgp/test_bgp_chasssis.py

```diff
@@ -0,0 +1,424 @@
+                              nexthops_with_inferior_paths=inferior_paths_nexthops,
+                              deleted=False)
+                for otherdut in otherduts:
+                    verify_routes(otherdut, routes[ipversion], [neigh_ip_addrs[bestpath_index]], [],
+                                  nexthops_with_inferior_paths=inferior_paths_nexthops,
+                                  deleted=False)
+
+                # Withdraw the bestpath and verify the alternate inferior path is chosen.
+                update_routes(exabgp_ports[bestpath_index], routes[ipversion], withdraw=True)
+                verify_routes(dut, routes[ipversion], inferior_paths_nexthops,
+                              [neigh_intfs_by_dut[dut][inferior_path_index]],
+                              nexthops_with_inferior_paths=None,
+                              deleted=False)
+                for otherdut in otherduts:
+                    verify_routes(otherdut, routes[ipversion], inferior_paths_nexthops, [],
+                                  nexthops_with_inferior_paths=None,
+                                  deleted=False)
+
+                # Withdraw the alternate path and verify the route is fully withdrawn from all
+                # the linecards.
+                update_routes(exabgp_ports[inferior_path_index], routes[ipversion], withdraw=True)
+                verify_routes(dut, routes[ipversion], [], [], nexthops_with_inferior_paths=None,
+                              deleted=True)
+                for duthost in duthosts:
+                    if duthost.is_supervisor_node():
+                        continue
+                    verify_routes(duthost, routes[ipversion], [], [],
+                                  nexthops_with_inferior_paths=None, deleted=True)
+
+            for ipversion in ipversions:
+                verify_bgp_inferior_path_propagation_af(ipversion)
+
+        def test_bgp_convergence_on_linkflap(dut, allduts):
+            neigh_intfs_by_dut = {IPV4:{}, IPV6:{}}
+            neigh_ip_addrs = {IPV4:{}, IPV6:{}}
+            def set_neigh_intfs_state(ifstate):
+                for ipversion in ipversions:
+                    for intf in neigh_intfs_by_dut[ipversion][dut]:
+                        set_peerintf_state(dut, intf, ifstate)
+
+            def verify_all_routes(deleted=False):
+                for ipversion in ipversions:
+                    verify_routes_af([dut], allduts, ipversion, neigh_ip_addrs[ipversion],
+                                     neigh_intfs_by_dut[ipversion], deleted=deleted)
+
+            for ipversion in ipversions:
+                neigh_intfs_by_dut[ipversion], neigh_ip_addrs[ipversion] = \
+                      update_and_verify_routes_af([dut], allduts, None, ipversion, False)
+
+            set_neigh_intfs_state('down')
+            verify_all_routes(deleted=True)
+            set_neigh_intfs_state('up')
+            verify_all_routes(deleted=False)
```

Comment:
preferably assert the return values within test function. It's easier to understand what is being tested. Is very helpful when the test fails.